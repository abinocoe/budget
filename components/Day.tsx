import React, { Component } from "react";
import { Text, TextInput, View } from "react-native";

interface Props {
  amountSpent: number;
  date: number;
  updateAmountSpent: (amount: number, day: number) => void;
}

class Day extends Component<Props> {
  public render() {
    return (
      <View
        style={{ flexDirection: "row", paddingVertical: 10 }}
        key={this.props.date}
      >
        <Text>{this.props.date}</Text>
        <TextInput
          keyboardType="number-pad"
          onChangeText={(text: any) => {
            if (!isNaN(text)) {
              this.props.updateAmountSpent(parseInt(text, 10), this.props.date);
            }
          }}
          style={{
            borderColor: "red",
            borderWidth: 1,
            width: "10%",
            marginHorizontal: 4
          }}
          value={
            isNaN(this.props.amountSpent)
              ? ""
              : this.props.amountSpent.toString()
          }
        />
      </View>
    );
  }
}

export default Day;
