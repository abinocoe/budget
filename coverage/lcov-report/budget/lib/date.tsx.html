<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for budget/lib/date.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">budget/lib</a> date.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/15</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import _ from "lodash";
&nbsp;
const today = new Date();
const dateToday = today.getDate();
&nbsp;
export const numberOfDaysInStartMonth = (startMonth: number) =&gt;
  32 -
  new Date(
    startMonth === 0 ? today.getFullYear() - 1 : <span class="branch-1 cbranch-no" title="branch not covered" >today.getFullYear(),</span>
    startMonth,
    32
  ).getDate();
&nbsp;
export const getDisplayDays = (startDate: number, startMonth: number) =&gt; {
  let days: number[];
  <span class="missing-if-branch" title="else path not taken" >E</span>if (dateToday &lt; startDate) {
    days = [
      ..._.range(startDate, numberOfDaysInStartMonth(startMonth) + 1).concat(
        dateToday === 1 ? <span class="branch-0 cbranch-no" title="branch not covered" >[1] </span>: _.range(1, dateToday + 1)
      )
    ];
  } else <span class="cstat-no" title="statement not covered" >if (dateToday === startDate) {</span>
<span class="cstat-no" title="statement not covered" >    days = [startDate];</span>
  } else {
<span class="cstat-no" title="statement not covered" >    days = _.range(startDate, dateToday + 1);</span>
  }
  return days;
};
&nbsp;
export const getPeriodStartMonth = (startDate: number) =&gt; {
  const thisMonth = today.getMonth();
  return dateToday &gt;= startDate
    ? <span class="branch-0 cbranch-no" title="branch not covered" >thisMonth</span>
    : thisMonth === 0
    ? <span class="branch-0 cbranch-no" title="branch not covered" >11</span>
    : thisMonth - 1;
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 13 2019 11:12:04 GMT+0000 (Greenwich Mean Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
