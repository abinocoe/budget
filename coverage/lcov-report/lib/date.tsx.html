<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/date.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">lib</a> date.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/15</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import _ from "lodash";
&nbsp;
const today = new Date();
const dateToday = today.getDate();
&nbsp;
export const numberOfDaysInStartMonth = <span class="fstat-no" title="function not covered" >(</span>startMonth: number) =&gt;
<span class="cstat-no" title="statement not covered" >  32 -</span>
  new Date(
    startMonth === 0 ? today.getFullYear() - 1 : today.getFullYear(),
    startMonth,
    32
  ).getDate();
&nbsp;
export const getDisplayDays = <span class="fstat-no" title="function not covered" >(</span>startDate: number, startMonth: number) =&gt; {
  let days: number[];
<span class="cstat-no" title="statement not covered" >  if (dateToday &lt; startDate) {</span>
<span class="cstat-no" title="statement not covered" >    days = [</span>
      ..._.range(startDate, numberOfDaysInStartMonth(startMonth) + 1).concat(
        dateToday === 1 ? [1] : _.range(1, dateToday + 1)
      )
    ];
  } else <span class="cstat-no" title="statement not covered" >if (dateToday === startDate) {</span>
<span class="cstat-no" title="statement not covered" >    days = [startDate];</span>
  } else {
<span class="cstat-no" title="statement not covered" >    days = _.range(startDate, dateToday + 1);</span>
  }
<span class="cstat-no" title="statement not covered" >  return days;</span>
};
&nbsp;
export const getPeriodStartMonth = <span class="fstat-no" title="function not covered" >(</span>startDate: number) =&gt; {
  const thisMonth = <span class="cstat-no" title="statement not covered" >today.getMonth();</span>
<span class="cstat-no" title="statement not covered" >  return dateToday &gt;= startDate</span>
    ? thisMonth
    : thisMonth === 0
    ? 11
    : thisMonth - 1;
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Feb 11 2019 10:11:00 GMT+0000 (Greenwich Mean Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
