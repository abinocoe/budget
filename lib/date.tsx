import _ from "lodash";

export const today = new Date();
export const dateToday = today.getDate();

const dayStrings = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];

export const numberOfDaysInStartMonth = (startMonth: number) =>
  32 -
  new Date(
    startMonth === 0 ? today.getFullYear() - 1 : today.getFullYear(),
    startMonth,
    32
  ).getDate();

export const getDisplayDays = (startDate: number, startMonth: number) => {
  let days: number[];
  if (dateToday < startDate) {
    days = [
      ..._.range(startDate, numberOfDaysInStartMonth(startMonth) + 1).concat(
        dateToday === 1 ? [1] : _.range(1, dateToday + 1)
      )
    ];
  } else if (dateToday === startDate) {
    days = [startDate];
  } else {
    days = _.range(startDate, dateToday + 1);
  }
  return days;
};

export const getPeriodStartMonth = (startDate: number) => {
  const thisMonth = today.getMonth();
  return dateToday >= startDate
    ? thisMonth
    : thisMonth === 0
    ? 11
    : thisMonth - 1;
};

export const getFriendlyDate = (day: number) => {
  const mutableDate = new Date();
  if (day > dateToday) {
    mutableDate.setDate(day);
    mutableDate.setMonth(today.getMonth() - 1);
  } else {
    mutableDate.setDate(day);
  }
  return `${dayStrings[mutableDate.getDay()]} ${day}${getOrdinal(day)}`;
};

const getOrdinal = (day: number) => {
  if (day > 3 && day < 21) return "th";
  switch (day % 10) {
    case 1:
      return "st";
    case 2:
      return "nd";
    case 3:
      return "rd";
    default:
      return "th";
  }
};
