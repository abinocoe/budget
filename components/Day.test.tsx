import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

// @ts-ignore
import { dateToday, today } from "../lib/date";

import Day from "./Day";

const currentDate = new Date();

afterEach(() => {
  today.setDate(currentDate.getDate());
  today.setMonth(currentDate.getMonth());
  today.setFullYear(currentDate.getFullYear());
});

describe("Day", () => {
  it("renders correctly on the 3rd of April 2019", () => {
    today.setMonth(3);
    today.setFullYear(2019);
    const output = shallow(
      <Day date={3} amountSpent={100} updateAmountSpent={() => null} />
    );
    expect(toJson(output)).toMatchSnapshot();
  });

  it("renders correctly on the 3rd of April 2019 with the input showing", () => {
    today.setMonth(3);
    today.setFullYear(2019);
    const output = shallow(
      <Day date={3} amountSpent={100} updateAmountSpent={() => null} />
    );
    output.setState({ editable: true });
    expect(toJson(output)).toMatchSnapshot();
  });
});
