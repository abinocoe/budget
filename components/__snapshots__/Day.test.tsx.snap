// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day renders correctly for the 3rd on the 4th of April 2019 1`] = `
<Themed.ListItem
  rightElement={
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "borderColor": "pink",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 44,
          "justifyContent": "center",
          "paddingRight": 5,
          "width": 100,
        }
      }
    >
      <Themed.Text
        onPress={[Function]}
        style={
          Object {
            "color": "#A9A9A9",
            "fontSize": 16,
          }
        }
      >
        1.00
      </Themed.Text>
    </View>
  }
  title="Wed 3rd"
/>
`;

exports[`Day renders correctly for the 3rd on the 4th of April 2019 with the input showing 1`] = `
<Themed.ListItem
  rightElement={
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "borderColor": "pink",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 44,
          "justifyContent": "center",
          "paddingRight": 5,
          "width": 100,
        }
      }
    >
      <ForwardRef(Themed.Input)
        autoFocus={true}
        containerStyle={
          Object {
            "paddingHorizontal": 0,
          }
        }
        inputContainerStyle={
          Object {
            "borderBottomWidth": 0,
          }
        }
        inputStyle={
          Object {
            "fontSize": 16,
            "marginRight": 0,
            "paddingRight": 0,
            "paddingVertical": 0,
            "textAlign": "right",
          }
        }
        keyboardType="numeric"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="1.00"
        placeholderTextColor="#A9A9A9"
        value=""
      />
    </View>
  }
  title="Wed 3rd"
/>
`;

exports[`Day renders correctly for the 5th on the 4th of April 2019 1`] = `
<Themed.ListItem
  rightElement={
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "borderColor": "pink",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 44,
          "justifyContent": "center",
          "paddingRight": 5,
          "width": 100,
        }
      }
    >
      <Themed.Text
        onPress={[Function]}
        style={
          Object {
            "color": "#A9A9A9",
            "fontSize": 16,
          }
        }
      >
        1.00
      </Themed.Text>
    </View>
  }
  title="Tue 5th"
/>
`;

exports[`Day renders correctly for the 5th on the 4th of April 2019 with the input showing 1`] = `
<Themed.ListItem
  rightElement={
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "borderColor": "pink",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 44,
          "justifyContent": "center",
          "paddingRight": 5,
          "width": 100,
        }
      }
    >
      <ForwardRef(Themed.Input)
        autoFocus={true}
        containerStyle={
          Object {
            "paddingHorizontal": 0,
          }
        }
        inputContainerStyle={
          Object {
            "borderBottomWidth": 0,
          }
        }
        inputStyle={
          Object {
            "fontSize": 16,
            "marginRight": 0,
            "paddingRight": 0,
            "paddingVertical": 0,
            "textAlign": "right",
          }
        }
        keyboardType="numeric"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="1.00"
        placeholderTextColor="#A9A9A9"
        value=""
      />
    </View>
  }
  title="Tue 5th"
/>
`;
